### DISCUSSION TITLE
Snapping  size is wrong for people with a side dock.

### DISCUSSION THREAD

**Discussion by CodingWithDavidZ** [rxhanson/Rectangle#602] [General]
- [ ] This is a new bug that hasn't been documented in a closed issue or in the Readme.
- [ ] Could we add an option in preferences to add padding to the left (or right) for people who use side dock?

macOS version:
Rectangle version:
Logs if applicable (In Rectangle menu, hold option, "View Logging..."):

**Comment by rxhanson** [comment]
If you’re using the macOS dock on the side, then this is likely a macOS bug that only happens immediately after the dock is hidden and displayed or a monitor is connected or disconnected. This is a bug that I don’t have plans to fix because it’s macOS side.

If you are not using a macOS dock and are instead using a third-party dock, then there are configurations that can specify gaps via terminal command. If this is the case then I am guessing this is a feature request for configuring this via the UI?

**Comment by rxhanson** [comment]
Converting this over to a discussion, since this isn't a Rectangle side bug and it appears to be a feature request for UI config.

**Comment by RyanRoberts** [comment]
I came here to bring up this suggestion but see it's already been made.

I've been using this app recently https://mightymac.app/intellidock/ that hides and shows the dock depending on available space. It would amazing if Rectangle Pro had an option to ignore the dock when sizing windows. It's just like the way Stage Manager works.

I'm not sure if anything has happened since your last comment you made in 21. The dev of Intellidock appears to have found a workaround.

**Reply by rxhanson** [reply]
Thanks for the additional feedback. The intellidock way to do it is kind of the only way I can think that this could work; I take it you're asking specifically for that feature? To make sure I understand correctly, the dock would get hidden if say, you maximize a window or give it a "left half" command with the dock on the left side?

**Reply by RyanRoberts** [reply]
Sorry I somehow overlooked CodingWithDavidZ is talking about a side dock. My dock is at the bottom so it works just like the Intellidock examples.

Adding the dock hiding feature would be great but I don't want to demand that much. At the moment with Rectangle and Intellidock if I make a window whatever size but full height, Rectangle sizes it with the dock in consideration but then Intellidock hides the dock leaving a space. I'd love it if Rectangle could somehow just ignore the dock. No problem if this is out of scope of what Rectangle is doing.
